src/viam/services/vision/vision.py

```diff
@@ -1,27 +1,106 @@
 
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
+
 from ..service_base import ServiceBase
 
 
+class CaptureAllResult:
+    """
+    CaptureAllResult represents the collection of things that you have requested from the
+    CaptureAllFromCamera method. This is used most often for visualization purposes, since normally,
+    returning the image on every call to a classifier/detector/etc would be costly and unnecessary.
+    The default result for each field is None rather than the empty list to distinguish between
+    "there was no request for the classifier/detector to return a result" vs.
+    "the classifier/detector was requested, but there were no results".
+    """
+    def __init__(self, image=None, classifications=None, detections=None, objects=None, extra={}):
+        """
+        Args:
+            image (ViamImage|None): The image from the GetImage request of the camera, if it was requested.
+            classifications (List[Classification]|None): The classifications from GetClassifications, if it was requested.
+            detections (List[Detection]|None): The detections from GetDetections, if it was requested.
+            objects (List[PointCloudObject]|None): the object point clouds from GetObjectPointClouds, if it was requested.
+            extra (dict): A catch all structure, usually for metadata, that a module writer might want to return. Default empty.
+
+        Returns:
+            None
+        """
+        self.image: Union[ViamImage, None] = image
+        self.detections: Union[List[Detection], None] = detections
+        self.classifications: Union[List[Classification], None] = classifications
+        self.objects: Union[List[PointCloudObject], None] = objects
+        self.extra: dict = extra
+
+
 class Vision(ServiceBase):
     """
     Vision represents a Vision service.
 
     This acts as an abstract base class for any drivers representing specific
-    arm implementations. This cannot be used on its own. If the ``__init__()`` function is
+    vision implementations. This cannot be used on its own. If the ``__init__()`` function is
     overridden, it must call the ``super().__init__()`` function.
     """
-
     SUBTYPE: Final = Subtype(  # pyright: ignore [reportIncompatibleVariableOverride]
         RESOURCE_NAMESPACE_RDK, RESOURCE_TYPE_SERVICE, "vision"
     )
 
+    Properties: "TypeAlias" = GetPropertiesResponse
+    """
+    Properties is a class that states what features are supported on the associated vision service.
+    Currently, these are the following properties:
+    classifications_supported (bool): GetClassifications and GetClassificationsFromCamera are implemented.
+    detections_supported (bool): GetDetections and GetDetectionsFromCamera are implemented.
+    object_point_clouds_supported (bool): GetObjectPointClouds is implemented.
+    """
```

Comment:
Will this show up in the python docs online?

Reply:
This hsould be fine. You might want to indent/add a `*` or `-` to the attributes

Reply:
linter actually complains about an unexpected indent, but I did add the `-`